### ISSUE TITLE
BERT Training

### ISSUE THREAD

**Issue by gpauloski** [gpauloski/llm-colossal-ai#1]
- [x] Config Files
  - [x] Define reference BERT phase 1/2
  - [x] Determine casing
  - [x] Test loading with branch statements in configs
  - [x] Infer `BATCH_SIZE` from `global_batch_size`
- [x] Configure logging
- [x] Data Loading
  - [x] Add [NVIDIA Dataset for BERT](https://github.com/hpcaitech/ColossalAI-Examples/blob/main/language/roberta/pretraining/nvidia_bert_dataset_provider.py)
- [x] Training Loss
- [x] Optimizers
  - [x] Support multiple types by name
  - [x] [Extract trainable params from model](https://github.com/hpcaitech/ColossalAI-Examples/blob/e0830ccc1bbc57f9c50bb1c00f3e23239bf1e231/language/roberta/pretraining/pretrain_utils.py#L77)
- [x] LR Scheduler
- [x] Checkpoint Saving
  - [x] Include optional kwargs for saving phase
- [x] Training Loop
- [x] TensorBoard
